src/boot/efi/boot.c

```diff
@@ -1431,10 +1430,6 @@ static VOID config_entry_add_from_file(
 
         entry->device = device;
         entry->id = StrDuplicate(file);
-        len = StrLen(entry->id);
-        /* remove ".conf" */
-        if (len > 5)
-                entry->id[len - 5] = '\0';
```

Comment:
There is matching code in `src/shared/bootspec.c:boot_entry_load`. This same change would have to be done there.